File: Java/1_TwoSum.java
// 1 - TwoSum

public int[] twoSum(int[] nums, int target) {
    Map<Integer, Integer> hash = new HashMap<>();
    for(int i = 0; i < nums.length; i++) {
        int c = target - nums[i];
        if(hash.containsKey(c)) {
            return new int[] { hash.get(c), i };
        }
        hash.put(nums[i], i);
    }
    return new int[] { -1, -1 };
}

File: CPP/1_TwoSum.cpp
// 1 - TwoSum

vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> hash;
    for(int i = 0; i < nums.size(); i++) {
        int c = target - nums[i];
        if(hash.find(c) != hash.end()) {
            return {hash[c], i};
        }
        hash[nums[i]] = i;
    }
    return {};
}